### PR TITLE
Add support for table and schema renames

### PR DESCRIPTION
**Pull request by ZygimantasKoncius**
This PR:
- Regenerates on the latest api-spec, meaning all previously unsupported operations are now supported(e.g. a bunch of data types)
- Adds support to schema and table updates - the operation is async in API, but we want to have it synchronous in TF to have consistent state, this is achieved by using `retry` util, the example of that working can be found below.
- I tagged our monkey-patches with `Etleap monkey-patch` comments, except for the plan modifier things - we'll collect them from diffs at later point
- The wait methods are also all in the pipeline_resource for now, we probably may want to split them out when we revisit automation, but i don't think it's necessary right now
- Additional files added to `.genignore`, because i needed to change `unmarshalJson` in those files, to allow for undeclared `schemaChangingTo`/`tableChangingTo` fields

```
Terraform will perform the following actions:

  # etleap_pipeline.test_pipeline will be updated in-place
  ~ resource "etleap_pipeline" "test_pipeline" {
      ~ destination                 = {
          ~ redshift = {
              ~ schema                   = "zygi_test" -> "zygi_dev"
              ~ table                    = "tf_test8" -> "tf_test10"
                # (7 unchanged attributes hidden)
            }
        }
        id                          = "uhyZPwBB"
        name                        = "TF Test pipeline"
        # (11 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

etleap_pipeline.test_pipeline: Modifying... [id=uhyZPwBB]
etleap_pipeline.test_pipeline: Still modifying... [id=uhyZPwBB, 10s elapsed]
etleap_pipeline.test_pipeline: Still modifying... [id=uhyZPwBB, 20s elapsed]
etleap_pipeline.test_pipeline: Still modifying... [id=uhyZPwBB, 30s elapsed]
etleap_pipeline.test_pipeline: Modifications complete after 31s [id=uhyZPwBB]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

Note to reviewer:
- Most of the files are auto-generated, all of my changes were in the files that are in `.genignore` now.

### CODE REVIEW

File: internal/provider/pipeline_resource.go [hunk 1]
```diff
@@ -55,7 +58,7 @@ type PipelineResource struct {
 type PipelineResourceModel struct {
 	CreateDate               types.String                         `tfsdk:"create_date"`
 	DeletionOfExportProducts types.Bool                           `tfsdk:"deletion_of_export_products"`
-	Destination              DestinationTypes                     `tfsdk:"destination"`
+	Destination              DestinationTypes                	  `tfsdk:"destination"`
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]: Formatting looks off,

File: internal/provider/pipeline_resource.go [hunk 2]
```diff
@@ -8643,11 +8671,12 @@ func (r *PipelineResource) Read(ctx context.Context, req resource.ReadRequest, r
 	}
 	data.RefreshFromSharedPipelineOutput(res.PipelineOutput)
 
+	// Etleap monkey-patch
 	existingDestination := data.Destination.Redshift
 	if existingDestination == nil {
-		data.Destination.Redshift = data.Destinations[0].Destination.Redshift
-		data.Destination.Snowflake = data.Destinations[0].Destination.Snowflake
-		data.Destination.DeltaLake = data.Destinations[0].Destination.DeltaLake
+		data.Destination.Redshift 	= data.Destinations[0].Destination.Redshift
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]: Same as above

File: internal/provider/pipeline_resource.go [hunk 3]
```diff
@@ -8587,11 +8614,12 @@ func (r *PipelineResource) Create(ctx context.Context, req resource.CreateReques
 	data.RefreshFromSharedPipelineOutput(res1.PipelineOutput)
 	refreshPlan(ctx, plan, &data, resp.Diagnostics)
 
+	// Etleap monkey-patch
 	existingDestination := data.Destination.Redshift
 	if existingDestination == nil {
-		data.Destination.Redshift = data.Destinations[0].Destination.Redshift
-		data.Destination.Snowflake = data.Destinations[0].Destination.Snowflake
-		data.Destination.DeltaLake = data.Destinations[0].Destination.DeltaLake
+		data.Destination.Redshift 	= data.Destinations[0].Destination.Redshift
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]: Formatting looks off

File: internal/provider/pipeline_resource.go [hunk 4]
```diff
@@ -8672,21 +8701,44 @@ func (r *PipelineResource) Update(ctx context.Context, req resource.UpdateReques
...
+	// Etleap monkey-patch end
 
 	request := operations.UpdatePipelineRequest{
 		ID:             id,
 		PipelineUpdate: pipelineUpdate,
 	}
 	res, err := r.client.Pipeline.Update(ctx, request)
 	if err != nil {
-		resp.Diagnostics.AddError("failure to invoke API", err.Error())
+		resp.Diagnostics.AddError("failure to invoke API eyo", err.Error())
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]:
```suggestion
		resp.Diagnostics.AddError("failure to invoke API", err.Error())
```

File: internal/provider/pipeline_resource.go [hunk 5]
```diff
@@ -8733,11 +8810,12 @@ func (r *PipelineResource) Update(ctx context.Context, req resource.UpdateReques
 	data.RefreshFromSharedPipelineOutput(res1.PipelineOutput)
 	refreshPlan(ctx, plan, &data, resp.Diagnostics)
 
+	// Etleap monkey-patch
 	existingDestination := data.Destination.Redshift
 	if existingDestination == nil {
-		data.Destination.Redshift = data.Destinations[0].Destination.Redshift
-		data.Destination.Snowflake = data.Destinations[0].Destination.Snowflake
-		data.Destination.DeltaLake = data.Destinations[0].Destination.DeltaLake
+		data.Destination.Redshift 	= data.Destinations[0].Destination.Redshift
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]: Formatting looks off.

File: internal/provider/pipeline_resource.go [hunk 6]
```diff
@@ -8791,3 +8869,75 @@ func (r *PipelineResource) Delete(ctx context.Context, req resource.DeleteReques
 func (r *PipelineResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), req.ID)...)
 }
+
+// Etleap monkey-patch functions
+func waitPipelineRenamed(ctx context.Context, r *PipelineResource, pipelineId string, destinationId string, originalSchema string, originalTable string, newSchema string, newTable string) (*operations.GetPipelineRequest, error) {
+	const (
+		timeout = 75 * time.Minute
```
Comment (caiusb) [inline internal/provider/pipeline_resource.go]: Do we expect this to take 75 minutes? Maybe we set it to something more reasonable, like 10-15 minutes?

**Review by caiusb — APPROVED** [review]
Looks great! +1 after comments!

File: internal/provider/pipeline_resource.go (hunk 6 above)
Comment (ZygimantasKoncius) [reply internal/provider/pipeline_resource.go]: Sure, made it 15 minutes.

**Pull request merged**